src/Achievements/Achievement.cs

```diff
@@ -19,14 +21,42 @@ public class Achievement : UniformTokenSplitter
 
 		protected Address Exchanger { get; set; }
 
+		protected string Describtion { get; set; }
+
+		protected IDictionary<Address, ITaggedTokens> TokenContributors { get; set; }
+			= new Dictionary<Address, ITaggedTokens>();
+
+		protected override void HandleMessage(Message message)
+		{
+			switch (message.Type)
+			{
+				case TokensReceivedEvent.Type:
+					var sender = message.Payload.Get<Address>(TokensReceivedEvent.From);
```

Comment:
Use explicit types

Reply:
The type (`Address`) is already present on the line.